### PR TITLE
Improve wifi open and close

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/net/wifi.toit [hunk 1]
```diff
@@ -15,26 +16,55 @@ import ..esp32
...
+      // connection. We always set it to a value when leaving this path
+      // and we always clear out the latch when we're done synchronizing.
+      wifi_connecting_ = monitor.Latch
+      wifi := wifi.Wifi
+      try:
+        wifi.set_ssid ssid password
+        with_timeout WIFI_CONNECT_TIMEOUT_: wifi.connect
+        with_timeout WIFI_DHCP_TIMEOUT_: wifi.get_ip
+        // Success: Register the WiFi connection, tell anyone who is waiting
+        // for it that the connetion is ready to be used (no exception), and
```
Comment (erikcorry) [inline lib/net/wifi.toit:47]:
```suggestion
        // for it that the connection is ready to be used (no exception), and
```

**Review by erikcorry — APPROVED** [review]

File: lib/net/wifi.toit (hunk 1 above)
Comment (floitsch) [inline lib/net/wifi.toit]:
```suggestion
        // for it that the connection is ready to be used (no exception), and
```

File: lib/net/wifi.toit [hunk 2]
```diff
@@ -15,26 +16,55 @@ import ..esp32
 WIFI_CONNECT_TIMEOUT_  ::= Duration --s=10
 WIFI_DHCP_TIMEOUT_     ::= Duration --s=16
 
+wifi_/wifi.Wifi? := null
+wifi_connecting_/monitor.Latch? := null
```
Comment (floitsch) [inline lib/net/wifi.toit:20]: Add a comment explaining what values are on this latch.
Maybe?
```suggestion
/**
Latch used to coordinate creation of the connection.
If the `get` is `null` the WiFi connection is established and can be used.
If it is not null, then it contains the error that was encountered during creation.
*/
wifi_connecting_/monitor.Latch? := null
```

File: lib/net/wifi.toit [hunk 3]
```diff
@@ -15,26 +16,55 @@ import ..esp32
...
+        wifi_connecting_.set null
+      finally: | is_exception exception |
+        if is_exception:
+          wifi.close
+          wifi_connecting_.set exception.value
+        wifi_connecting_ = null
+
+  interface := WifiInterface_
+  if ssid == wifi_.ssid_ and password == wifi_.password_: return interface
+  interface.close
```
Comment (floitsch) [inline lib/net/wifi.toit:59]: Why even allocate the interface if we are not going to use it?

Would be simpler to just:
```
if ssid == wifi_.ssid_ and password == wifi_.password_: return WifiInterface_
```

**Review by floitsch — COMMENTED** [review]
LGTM.

File: lib/net/wifi.toit (hunk 3 above)
Comment (kasperl) [reply lib/net/wifi.toit:59]: The wifi_ isn't closed.

**Pull request merged**